solidity/dashboard/src/components/WithdrawETHModal.jsx

```diff
@@ -0,0 +1,157 @@
+import React, { useCallback } from "react"
+import { SubmitButton } from "./Button"
+import { useCustomOnSubmitFormik } from "../hooks/useCustomOnSubmitFormik"
+import FormInput from "./FormInput"
+import { getErrorsObj } from "../forms/common-validators"
+import { withFormik } from "formik"
+import { colors } from "../constants/colors"
+import web3Utils from "web3-utils"
+import { useWeb3Context } from "./WithWeb3Context"
+import { tbtcAuthorizationService } from "../services/tbtc-authorization.service"
+import { useShowMessage, messageType } from "./Message"
+import * as Icons from "./Icons"
+
+const AvailableEthCell = React.memo(({ availableETH }) => {
+  return (
+    <>
+      <span
+        className="text-big text-grey-70"
+        style={{
+          textAlign: "right",
+          padding: "0.25rem 1rem",
+          paddingLeft: "2rem",
+          borderRadius: "100px",
+          border: `1px solid ${colors.grey20}`,
+          backgroundColor: `${colors.grey10}`,
+        }}
+      >
+        {availableETH}
+      </span>
+      <span style={{ color: `${colors.grey60}` }}>&nbsp;ETH</span>
+    </>
+  )
+})
+
+const WithdrawETHModal = ({ operatorAddress, availableETH, closeModal }) => {
```

Comment:
I'm wondering if we can extract a common part of this file and `AddETHModal.jsx`. Maybe as a first step let's create a common `Form` component which will be look like that: 

```
const Form = ({ validationFn, onSubmit, closeModal, ...formikProps }) => {
  const onSubmitBtn = useCustomOnSubmitFormik(onSubmit)

  return (
    <form>
      <FormInput
        name="ethAmount"
        type="text"
        label="ETH Amount"
        placeholder="0"
      />
      <div
        className="flex row center mt-2"
        style={{
          borderTop: `1px solid ${colors.grey20}`,
          margin: "0 -2rem",
          padding: "2rem 2rem 0",
        }}
      >
        <SubmitButton
          className="btn btn-primary"
          type="submit"
          onSubmitAction={onSubmitBtn}
          withMessageActionIsPending={false}
          triggerManuallyFetch={true}
          disabled={!formikProps.dirty}
        >
          add eth
        </SubmitButton>
        <span onClick={closeModal} className="ml-1 text-link">
          Cancel
        </span>
      </div>
    </form>
  )
}
```

and then
```
withFormik({
  validateOnChange: false,
  validateOnBlur: false,
  mapPropsToValues: () => ({
    ethAmount: "0",
  }),
  validate: (values, { validationFn }) => {
    return validationFn(values)
  }
})(Form)
```

Reply:
@r-czajkowski something is not right here after I've [extracted](https://github.com/keep-network/keep-core/pull/1825/commits/384babe4e663e10fef9e3dab9b42f2738b4f8fc9) the code. Will need your pair of eyes.

Reply:
already fixed :)